src/main/java/com/aws/greengrass/lifecyclemanager/KernelAlternatives.java

```diff
@@ -392,6 +417,91 @@ public void cleanupLaunchDirectoryLinks() {
+            // Exclude components that are not explicitly configured to bootstrap-on-rollback
+            Set<String> unconfiguredComponents = getComponentsNotConfiguredToBootstrapOnRollback(rollbackConfig);
+            logger.atDebug().kv("components", unconfiguredComponents)
+                    .log("These components are not configured to execute bootstrap steps during rollback. "
+                            + "They will be excluded from bootstrap-on-rollback.");
+            componentsToExclude.addAll(unconfiguredComponents);
+            bootstrapOnRollbackRequired = bootstrapManager.isBootstrapRequired(rollbackConfig.toPOJO(),
+                    componentsToExclude);
+        } catch (ServiceUpdateException | ComponentConfigurationValidationException exc) {
+            logger.atError().log("Rollback config invalid or could not be parsed", exc);
+            return false;
+        }
+        Path rollbackBootstrapTaskFilePath;
+        try {
+            rollbackBootstrapTaskFilePath = deploymentDirectoryManager.getRollbackBootstrapTaskFilePath();
+        } catch (IOException exc) {
+            logger.atError().log("Bootstrap-on-rollback task file path could not be resolved", exc);
+            return false;
+        }
+        if (bootstrapOnRollbackRequired) {
+            // Bootstrap-on-rollback is required, so write the task file.
+            try {
+                bootstrapManager.persistBootstrapTaskList(rollbackBootstrapTaskFilePath);
+            } catch (IOException exc) {
+                logger.atError().log("Bootstrap-on-rollback task file could not be written", exc);
+                return false;
+            }
+        } else {
+            // Bootstrap-on-rollback is not required, so ensure that the task file is deleted.
+            try {
+                bootstrapManager.deleteBootstrapTaskList(rollbackBootstrapTaskFilePath);
+            } catch (IOException exc) {
+                logger.atError().log("Bootstrap-on-rollback task file could not be cleaned up", exc);
+                return false;
+            }
+        }
+        return bootstrapOnRollbackRequired;
+    }
+
+    private Set<String> getComponentsNotConfiguredToBootstrapOnRollback(Configuration rollbackConfig) {
+        Topics services = rollbackConfig.findTopics(SERVICES_NAMESPACE_TOPIC);
+        if (services == null) {
+            return Collections.emptySet();
+        }
+        Set<String> componentsNotConfiguredToBootstrapOnRollback = new HashSet<>();
+        services.forEach((service) -> {
+            String serviceName = service.getName();
+            if (service instanceof Topics) {
+                boolean bootstrapOnRollback = Coerce.toBoolean(((Topics) service).findOrDefault(false,
+                        SERVICE_LIFECYCLE_NAMESPACE_TOPIC, LIFECYCLE_BOOTSTRAP_NAMESPACE_TOPIC,
+                        BOOTSTRAP_ON_ROLLBACK_CONFIG_KEY));
+                if (! bootstrapOnRollback) {
```

Comment:
nit: white space

Reply:
Oh, my style is to do this deliberately... it draws the eye to the NOT operator.

Reply:
But for you, I will remove the space :)